types/cli-color/columns.d.ts

```diff
@@ -0,0 +1,29 @@
+declare namespace columns {
+    export interface ColumnOptions {
+        /**
+         * align: Possible options: `'left'`, `'right'` (defaults to `'left'`)
+         */
+        align?: 'left' | 'right';
+    }
+
+    export interface ColumnsOptions {
+        /**
+         *  Custom colums separator (defaults to `|`)
+         */
+        sep?: string;
+        /**
+         * columns: Per column customizations, as e.g. `[{ align: 'right' }, null, { align: 'left' }]`
+         */
+        columns?: Array<ColumnOptions | null>;
+    }
+
+    export type Row = Iterable<any> | ArrayLike<any>;
+    export type Data = Iterable<Row> | ArrayLike<Row>;
+}
+
+/**
+ * Outputs aligned table of columns.
+ */
+declare function columns(data: ReadonlyArray<ReadonlyArray<any>>, options?: columns.ColumnsOptions): string;
+declare function columns(data: columns.Data, options?: columns.ColumnsOptions): string;
```

Comment:
Could combine these two overloads using

```ts
declare function columns(data: columns.Data | ReadonlyArray<ReadonlyArray<any>>, ...
```

though I'm not sure if that makes the types any easier to use.

Reply:
In fact, `ReadonlyArray<ReadonlyArray<any>>` fits signature of `columns.Data`, added it just for a hint to use arrays for `columns.Data`, should I just remove it?

Reply:
Oh, true. I'm probably not the best person to ask which way is better. Probably depends on how confusing the second overload looks in code completion.

Reply:
Is recommended to use union types: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-union-types

Also put more specific types first (this is for overloads, I'm not sure if this will also apply for types) ://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#ordering

Reply:
I'll merge as is. Any improvements could be handled by another PR.